### PR TITLE
py-onnx: Update to version 1.16.1

### PR DESCRIPTION
**Pull request by essandess**
Depends upon #24096, #24107

###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS x.y
Xcode x.y / Command Line Tools x.y.z

###### Verification 
Have you

- [ ] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [ ] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [ ] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL in commit message? 
- [ ] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?
- [ ] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

File: python/py-onnx/Portfile [hunk 1]
```diff
@@ -4,40 +4,39 @@ PortSystem          1.0
...
+checksums           md5     07fe47dc50e802e4a9051de9b3cdf928 \
+                    rmd160  e9817f10bbce05a2a7d5a716c7557467637c5287 \
+                    sha256  8299193f0f2a3849bfc069641aa8e4f93696602da8d165632af8ee48ec7556b6 \
+                    size    12306956
+
+python.versions     38 39 310 311 312
 
 if {${name} ne ${subport}} {
 
     compiler.cxx_standard 2011
```
Comment (reneeotten) [inline python/py-onnx/Portfile:35]: are you sure it doesn't require `2017` now; that's the only thing mentioned in the `CMakeLists.txt`

File: python/py-onnx/Portfile [hunk 2]
```diff
@@ -4,40 +4,39 @@ PortSystem          1.0
...
 
     depends_build-append \
         path:bin/cmake:cmake \
+        port:py${python.version}-cmake \
```
Comment (reneeotten) [inline python/py-onnx/Portfile:39]: I don't see why this would be necessary - what's the problem with using the already existing `cmake` port?

**Review by reneeotten — CHANGES_REQUESTED** [review]

File: python/py-onnx/Portfile (hunk 1 above)
Comment (essandess) [reply python/py-onnx/Portfile:35]: Fixed, thanks.

File: python/py-onnx/Portfile [hunk 3]
```diff
@@ -4,40 +4,39 @@ PortSystem          1.0
...
         port:py${python.version}-pytest-runner \
```
[the remark this answers is not in the document]
Comment (essandess) [reply python/py-onnx/Portfile:40]: It's in there somewhere. Without this dependency, this build error is thrown:

```sh
:info:build writing manifest file 'onnx.egg-info/SOURCES.txt'
:info:build ERROR Missing dependencies:
:info:build     cmake
:info:build Command failed:  cd "/opt/local/var/macports/build/_opt_local_ports_python_py-onnx/py312-onnx/work/onnx-1.16.1" && /opt/local/Library/Frameworks/Python.framework/Versions/3.12/bin/python3.12 -m build --no-isolation --wheel --outdir /opt/local/var/macports/build/_opt_local_ports_python_py-onnx/py312-onnx/work 
:info:build Exit code: 1
```

File: python/py-onnx/Portfile (hunk 2 above)
Comment (essandess) [reply python/py-onnx/Portfile:39]: Without this dependency, this build error is thrown:

```sh
:info:build writing manifest file 'onnx.egg-info/SOURCES.txt'
:info:build ERROR Missing dependencies:
:info:build     cmake
:info:build Command failed:  cd "/opt/local/var/macports/build/_opt_local_ports_python_py-onnx/py312-onnx/work/onnx-1.16.1" && /opt/local/Library/Frameworks/Python.framework/Versions/3.12/bin/python3.12 -m build --no-isolation --wheel --outdir /opt/local/var/macports/build/_opt_local_ports_python_py-onnx/py312-onnx/work 
:info:build Exit code: 1
```

Comment (essandess) [reply python/py-onnx/Portfile]: Okay, thank you. I've done that, and it works.

**Pull request merged**